CIPs/0010.md

```diff
@@ -29,15 +29,36 @@ As applications get developed on Celo, it becomes evident that this mechanism of
 
 Expand the `Signers` struct in `Accounts.sol` with a generic mapping and generic version of `authorizeXSigner`, `removeXSigner`, `getXSigner` and `signerToAccount` that take as additional input `string namespace`.
 
-- multiple account keys? (effectively making lockeedgold somewhat of a custody solution)
-- authorization key
-- makes key management easier by allowing tools implement once interface
+- Multiple account keys? (effectively making locked gold somewhat of a custody solution)
+- Authorization key
+- Makes key management easier by allowing tools implement once interface
 
 ## Alternative Solutions
 
-Alternately, applications could always derive their own authorization scheme/smart contracts. However, it would not benefit of the tooling that could be possible to faciliate key management.
+Alternately, applications could always derive their own authorization scheme/smart contracts. However, it would not benefit from the tooling that could be possible for facilitating key management.
 
-Instead of extending the `Signers` struct, a completely different smart contract `AuthorizedSigners` could be created. Functionality would be practically duplicated, but separated from the critical PoS system while gaining the benefit from a standardized authorizaton scheme.
+Instead of extending the `Signers` struct, a completely different smart contract `AuthorizedSigners` could be created. Functionality would be practically duplicated, but separated from the critical PoS system while gaining the benefit from a standardized authorization scheme.
+
+Alternative Solution: Enable smart contracts to be used as signers (as an addition to existing implementation)
```

Comment:
I probably wouldn't say that this is an alternative solution, IMO it should be part of the Proposed solution. Alternative solutions are other solutions that have been explored but are not recommended for various reasons

Reply:
I see, thanks for clearing that up Nam, I'll move it now.